### PR TITLE
Software Installation tests and framework - fixes #2055

### LINKED ISSUE #2055: Implement Eclipse Software Installation tests
This test should cover 'Install New Software' options and allow to test software installation from update site. 

Will be part of tests.
Will consist of test-framework - implementation of necessary dialogs, wait conditions, pages and other object APIs.
Test-framework will be exported as thus, it will be possible to use it externally
Test covering installation of software
RedDeer CI job that will be run after any snapshot/milestone/release (other issue)

### PR DESCRIPTION
**Pull request by odockal**
Signed-off-by: Ondrej Dockal <odockal@redhat.com>

# Pull Request Checklist

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?
* Is copyright with correct year in new files?
* Is there a JavaDoc in every class with description of a class and an author(and issue if suitable)?
* Is there a link to github project issue? (if needed)

## Testing
* Does it work?
* Are the tests green on Jenkins? Is there a link to Jenkins build?

### CODE REVIEW

**Comment by odockal** [comment]
@jkopriva ready to go. Can you please check it out?

**Comment by odockal** [comment]
@jkopriva  https://ci.eclipse.org/reddeer/job/reddeer.installation.test-pipeline/7/

File: tests/org.eclipse.reddeer.integration.test/src/org/eclipse/reddeer/integration/test/installation/common/dialog/InstallNewSoftwareDialog.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+		new WaitWhile(new WindowIsAvailable(this), timeout);
+		// can not wait in some cases
+		/*
+		try{
+		  
+			new WaitWhile(new JobIsRunning(), timeout);
+		} catch (NoClassDefFoundError e) {
+			// do nothing, reddeer.workbench plugin is not available
+		}*/
```
Comment (jkopriva) [inline tests/org.eclipse.reddeer.integration.test/src/org/eclipse/reddeer/integration/test/installation/common/dialog/InstallNewSoftwareDialog.java]: is this needed or can we delete this commented code?

Comment (odockal) [reply tests/org.eclipse.reddeer.integration.test/src/org/eclipse/reddeer/integration/test/installation/common/dialog/InstallNewSoftwareDialog.java]: Deleted.

**Comment by jkopriva** [comment]
LGTM, lets merge it!

**Pull request merged**